front/components/items/FindOrCancelAppointment/index.tsx

```diff
@@ -0,0 +1,198 @@
+import { useFormik } from 'formik';
+import React, { useEffect, useRef, useState } from 'react';
+import * as Yup from 'yup';
+import Paragraph from '../Paragraph';
+import Button from '../Button';
+import { strapiAppointmentGet } from 'lib/strapi/appointments/get';
+import { filtersAppointmentCode } from './models';
+import { strapiAppointmentQuery } from 'lib/strapi/appointments/queryType';
+import { strapiAppointmentPut } from 'lib/strapi/appointments/put';
+import WrapperColumn from 'components/wrappers/WrapperColumn';
+
+const FindOrCancelAppointment = () => {
+    const [data, setData] = useState<string>();
+    const [isSubmited, setIsSubmited] = useState(false);
+    const [isSearching, setIsSearching] = useState(false);
+    const [isCanceling, setIsCanceling] = useState(false);
+    const [messageOfCancelingProperlyDone, setMessageOfCancelingProperlyDone] = useState<string>();
+    const properCodeInput = useRef<HTMLInputElement>(null);
+    const [foundAppointment, setFoundAppoinment] = useState<strapiAppointmentQuery[]>();
+
+    const filters: filtersAppointmentCode = {
+        appointment_code: {
+            $eq: data,
+        },
+    };
+
+    const validateCode = () => {
+        if (properCodeInput.current)
+            properCodeInput.current.value = properCodeInput.current.value.replace(/[^0-9]/g, '');
+    };
+
+    const canceledAppointment = {
+        data: {
+            appointment_code: null,
+            is_booked: false,
+        },
+    };
+
+    const canceledAppointmentId = () => {
+        if (foundAppointment?.length === 1 && isCanceling) {
+            return foundAppointment[0].id.toString();
+        }
+    };
+
+    const formik = useFormik({
+        initialValues: {
+            code: '',
+        },
+        validationSchema: Yup.object().shape({
+            code: Yup.string()
+                .length(6, 'Kod powinien posuadać 6 cyfr')
+                .matches(/^[0-9]*$/, 'Podaj kod w cyfrach')
+                .required('Wymagane'),
+        }),
+        onSubmit: async () => {
+            setData(formik.values.code);
+            canceledAppointmentId();
+            formik.resetForm({ values: { code: '' } });
+            setIsSubmited(true);
+        },
+    });
+
+    useEffect(() => {
+        strapiAppointmentGet({ filters: filters, population: 'populate=*' }).then((res) => {
+            setFoundAppoinment(res.data.data);
+        });
+    }, [data || isSubmited]);
+
+    useEffect(() => {
+        const canceledAppointmentIdResult = canceledAppointmentId();
+
+        const cancellingAppointment = async () => {
+            if (isCanceling && canceledAppointmentIdResult) {
```

Comment:
to prevent too much wrapping
```suggestion
            if (!isCanceling || !canceledAppointmentIdResult) return;
```